azdev/operations/command_change/util.py

```diff
@@ -0,0 +1,217 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+
+import json
+import os
+import re
+from enum import Enum
+import jsbeautifier
+from knack.log import get_logger
+
+logger = get_logger(__name__)
+
+CMD_NAME_PATTERN = re.compile(r"\[\'commands\'\]\[\'([a-zA-Z0-9\-\s]+)\'\]")
+CMD_PARAMETER_PROPERTY_PATTERN = re.compile(r"\[(.*?)\]")
+
+
+class ChangeType(int, Enum):
+    DEFAULT = 0
+    ADD = 1
+    CHANGE = 2
+    REMOVE = 3
+
+
+def get_command_tree(command_name):
+    """
+    input: monitor log-profiles create
+    ret:
+    {
+        is_group: True,
+        group_name: 'monitor',
+        sub_info: {
+            is_group: True,
+            group_name: 'monitor log-profiles',
+            sub_info: {
+                is_group: False,
+                cmd_name: 'monitor log-profiles create'
+            }
+        }
+    }
+    """
+    name_arr = command_name.split()
+    ret = {}
+    name_arr.reverse()
+    for i, _ in enumerate(name_arr):
+        tmp = {}
+        if i == 0:
+            tmp = {
+                "is_group": False,
+                "cmd_name": " ".join(name_arr[::-1])
+            }
+        else:
+            tmp = {
+                "is_group": True,
+                "group_name": " ".join(name_arr[len(name_arr): (i - 1): -1]),
+                "sub_info": ret
+            }
+        ret = tmp
+    return ret
+
+
+def process_aaz_argument(az_arguments_schema, argument_settings, para):
+    from azure.cli.core.aaz import has_value  # pylint: disable=import-error
+    _fields = az_arguments_schema._fields  # pylint: disable=protected-access
+    aaz_type = _fields.get(argument_settings["dest"], None)
+    if aaz_type:
+        para["aaz_type"] = aaz_type.__class__.__name__
+        para["type"] = aaz_type._type_in_help  # pylint: disable=protected-access
+        if has_value(aaz_type._default):  # pylint: disable=protected-access
+            para["aaz_default"] = aaz_type._default  # pylint: disable=protected-access
+        if para["aaz_type"] in ["AAZArgEnum"] and aaz_type.get("enum", None) and aaz_type.enum.get("items", None):
+            para["aaz_choices"] = aaz_type.enum["items"]
```

Comment:
It seems not retrieve the first level properties of `AAZObjectArg` or `AAZListArg` with object elements.

Reply:
Right, aaz cmds has more details needs to be covered and detecting and verifying its changes. For now, only `choices` and `default` are supported.